assertEquals(set,set) now requires same ordering which it didn't before

In  TestNG 7.4.0(and all previous versions?) the following example worked fined and both sets where considered equal.
        HashSet<Integer> set1 = new LinkedHashSet<>();
        HashSet<Integer> set2 = new LinkedHashSet<>(); //same values different order.
        set1.add(1);
        set1.add(2);
        set2.add(2);
        set2.add(1);
        Assert.assertEquals(set1, set2);

the AbstractSet.equals() implementation in Java and used by most Set implementations define equality as the two sets containing the same elements but not necessarily in the same order. 

on the latest master of TestNG however the above example no longer works and throws:
 "java.lang.AssertionError: Iterators not same element order: expected: 1 and actual: 2"

the line causing this issue is: 
https://github.com/cbeust/testng/blob/cd85bcb7afebbaecd2299cbc72632e27f89a441b/testng-asserts/src/main/java/org/testng/Assert.java#L1459
which now iterates both sets and require the same ordering. 

as Objects.equals(actual, expected) is already know to be true it would have made more sense to just return true instead. (so it functions as before.
It also makes more sense for TestNGs assertEquals(set,set) to align with Java's definition of set.equals(set) as it was before. 

also even the comment in the same methods implies that you don't want to check ordering: 
https://github.com/cbeust/testng/blob/cd85bcb7afebbaecd2299cbc72632e27f89a441b/testng-asserts/src/main/java/org/testng/Assert.java#L1447

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+Fixed: GITHUB-2643: assertEquals(Set,Set) now ignores ordering as it did before. (Elis Edlund)
 Fixed: GITHUB-2653: Assert methods requires casting since TestNg 7.0 for mixed boxed and unboxed primitives in assertEquals.
 Fixed: GITHUB-2229: Restore @BeforeGroups and @AfterGroups Annotations functionality (Krishnan Mahadevan)
 Fixed: GITHUB-2563: Skip test if its data provider provides no data (Krishnan Mahadevan)

--- a/testng-asserts/src/main/java/org/testng/Assert.java
+++ b/testng-asserts/src/main/java/org/testng/Assert.java
@@ -1795,7 +1795,7 @@ public class Assert {
   }
 
   private static String getNotEqualReason(Set<?> actual, Set<?> expected) {
-    if (actual == expected) { // We don't use Arrays.equals here because order is checked
+    if (actual == expected) {
       return null;
     }
 
@@ -1807,7 +1807,7 @@ public class Assert {
     if (!Objects.equals(actual, expected)) {
       return "Sets differ: expected " + expected + " but got " + actual;
     }
-    return getNotEqualReason(actual.iterator(), expected.iterator());
+    return null;
   }
 
   /**

--- a/testng-asserts/src/test/java/test/asserttests/AssertTest.java
+++ b/testng-asserts/src/test/java/test/asserttests/AssertTest.java
@@ -373,8 +373,8 @@ public class AssertTest {
     assertNotEquals(iterator1, iterator2);
   }
 
-  @Test(description = "GITHUB-2540", expectedExceptions = AssertionError.class)
-  public void checkSetEqualsFailsWhenDifferentOrder() {
+  @Test(description = "GITHUB-2643")
+  public void checkSetEqualsWhenDifferentOrder() {
 
     Set<String> set1 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
     Set<String> set2 = new LinkedHashSet<>(Arrays.asList("a", "c", "b"));
@@ -400,8 +400,8 @@ public class AssertTest {
     assertEquals(set1, set2);
   }
 
-  @Test(description = "GITHUB-2540")
-  public void checkSetNotEquals() {
+  @Test(description = "GITHUB-2643", expectedExceptions = AssertionError.class)
+  public void checkSetNotEqualsFailsWhenDifferentOrder() {
 
     Set<String> set1 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
     Set<String> set2 = new LinkedHashSet<>(Arrays.asList("a", "c", "b"));
@@ -409,13 +409,31 @@ public class AssertTest {
     assertNotEquals(set1, set2);
   }
 
-  @Test(description = "GITHUB-2540", expectedExceptions = AssertionError.class)
-  public void checkSetNotEqualsFailsWhenDifferentOrder() {
+  @Test(description = "GITHUB-2643", expectedExceptions = AssertionError.class)
+  public void checkSetNotEqualsFailsWhenSameOrder() {
 
     Set<String> set1 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
     Set<String> set2 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
 
     assertNotEquals(set1, set2);
+  }
+
+  @Test(description = "GITHUB-2643")
+  public void checkSetNotEqualsWhenNotSameSets() {
+
+    Set<String> set1 = new LinkedHashSet<>(Arrays.asList("a", "b"));
+    Set<String> set2 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
+
+    assertNotEquals(set1, set2);
+  }
+
+  @Test(description = "GITHUB-2643", expectedExceptions = AssertionError.class)
+  public void checkSetEqualsFailsWhenNotSameSets() {
+
+    Set<String> set1 = new LinkedHashSet<>(Arrays.asList("a", "b", "c"));
+    Set<String> set2 = new LinkedHashSet<>(Arrays.asList("a", "b"));
+
+    assertEquals(set1, set2);
   }
 
   @Test(description = "GITHUB-2540", expectedExceptions = AssertionError.class)

